### ISSUE TITLE
Upstream Snabb NFV into OpenStack Juno

### ISSUE THREAD

**Issue by lukego** [SnabbCo/snabbswitch#101]
Ensure that [OpenStack Juno](https://www.openstack.org/software/roadmap/) supports Snabb NFV "out of the box".

**Comment by virtuallynathan** [comment]
Does this being close imply that Juno onwards does support Snabb NFV out of the box?

**Comment by lukego** [comment]
Hi Nathan,

No. Currently we are compatible with Icehouse and require a small number of patches to Nova and Neutron. OpenStack is gradually taking in NFV related features and we are being patient and maintaining our code downstream.

The holdup is that NFV functionality in general and vhost-user in particular is taking time to be accepted by OpenStack even though it is already in QEMU and Libvirt. We submitted patches to Juno but those slipped. Intel have kindly taken over and resubmitted this to Kilo and hopefully that will go in.

The good news is that we have reduced our total OpenStack changes down to about 200 lines of simple code that we can maintain downstream for as long as is needed.

Here are our latest Git trees for nova/neutron/qemu: https://github.com/SnabbCo/snabb-nfv

And here is the upstreaming effort now being lead by Intel: http://specs.openstack.org/openstack/nova-specs/specs/kilo/approved/libvirt_vif_vhostuser.html
